### PR TITLE
Address issues found in the audit of the compile flags

### LINKED ISSUE #1238: pkg-config: Validate compile/link flags
When building test enclaves, we build using the following flags  (as of this writing):

-g   -Wall -Werror -Wpointer-arith -Wconversion -Wextra -Wno-missing-field-initializers -fno-strict-aliasing -mxsave -fno-builtin-malloc -fno-builtin-calloc -m64 -fPIC -nostdinc -fno-stack-protector -fvisibility=hidden -nostdinc++ -std=c++14

Not all of these flags are exposed by pkg-config.

Additionally, we use the -pie linker option. The [documentation ](https://gcc.gnu.org/onlinedocs/gcc-8.2.0/gcc/Link-Options.html)for that says:
**-pie**
Produce a dynamically linked position independent executable on targets that support it. F**or predictable results, you must also specify the same set of options used for compilation** (-fpie, -fPIE, or model suboptions) when you specify this linker option.

CC: @mikbras @andschwa

### PR DESCRIPTION
**Pull request by andschwa**
This resolves #1238 (and makes it easier to maintain these lists of flags).

### CODE REVIEW

**Comment by andschwa** [comment]
bors try

File: 3rdparty/mbedtls/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+  -DOE_NEED_STDC_NAMES
+  -fPIE
+  -ftls-model=local-exec
+  -fvisibility=hidden
+  -fno-stack-protector
+  -fno-builtin-udivti3)
 
 if (USE_CLANGW)
-  set(MBEDTLS_WRAP_CFLAGS "-target x86_64-pc-linux ${MBEDTLS_WRAP_CFLAGS}")
+  list(APPEND MBEDTLS_WRAP_CFLAGS_LIST "-target x86_64-pc-linux")
```
Comment (andschwa) [inline 3rdparty/mbedtls/CMakeLists.txt]: @anakrish I don't have a Windows machine with which to test this. It really ought not to be a problem, but does the CI run validate this well enough?

File: cmake/compiler_settings.cmake [hunk 2]
```diff
@@ -51,12 +51,14 @@ include(CheckCCompilerFlag)
 include(CheckCXXCompilerFlag)
 
 # Apply Spectre mitigations if available.
-set(SPECTRE_MITIGATION_FLAGS "-mllvm;-x86-speculative-load-hardening")
+set(SPECTRE_MITIGATION_FLAGS -mllvm -x86-speculative-load-hardening)
```
Comment (andschwa) [inline cmake/compiler_settings.cmake]: This is a CMake list, so I removed some redundancy.

File: pkgconfig/CMakeLists.txt [hunk 3]
```diff
@@ -21,38 +21,47 @@ set(ENCLAVE_CINCLUDES
...
-string(APPEND ENCLAVE_CLIBS_1 "-lmbedx509 ")
-string(APPEND ENCLAVE_CLIBS_1 "-lmbedcrypto ")
-
-string(APPEND ENCLAVE_CLIBS_2 "-loelibc ")
-string(APPEND ENCLAVE_CLIBS_2 "-loecore ")
-
-set(ENCLAVE_CLIBS "${ENCLAVE_CLIBS_1} ${ENCLAVE_CLIBS_2}")
-
-set(ENCLAVE_CXXLIBS "${ENCLAVE_CLIBS_1} -loelibcxx ${ENCLAVE_CLIBS_2}")
+set(ENCLAVE_CLIBS_1
```
Comment (andschwa) [inline pkgconfig/CMakeLists.txt]: Sorry but I didn't want to maintain proper spacing 😄

File: 3rdparty/mbedtls/CMakeLists.txt (hunk 1 above)
Comment (CodeMonkeyLeet) [reply 3rdparty/mbedtls/CMakeLists.txt]: CI should be validating this already as part of the WS2016 configuration.

Comment (andschwa) [reply 3rdparty/mbedtls/CMakeLists.txt]: Thanks!

**Pull request merged**